src/storage_credential.cpp

```diff
@@ -80,4 +80,17 @@ namespace azure {  namespace storage_lite {
         std::string transformed_url = transform_url(h.get_url());
         h.set_url(transformed_url);
     }
+
+    AZURE_STORAGE_API token_credential::token_credential(const std::string &token)
+        : m_token(std::move(token)) {}
+
+    void token_credential::set_token(const std::string& token) {
+        m_token = token;
```

Comment:
Can you please make this thread safe? There is no guarantee that user will set the token and sign the request in the same thread.

Reply:
Does this satisfy everything here? Forgive me as I’ve done very little threaded c++

Reply:
I am afraid not. To make it thread safe you would need to add a mutex to guard the setters/getters so that the `m_token` will never be consumed while it's in the middle of a state change.

Reply:
The top answer in [this link](https://www.quora.com/What-should-a-C++-thread-safe-class-be-like) is a very good place to start with.

Reply:
@katmsft I removed the getter as it wasn't used and using a `std::lock_guard` in the setter and use of `m_token` in the `sign_request` method as well. 
That Quora answer also had `std::exchange`, but that's not available until c++14 it seems